tex/generic/pgf/frontendlayer/tikz/libraries/tikzlibrarydecorations.code.tex

```diff
@@ -17,8 +17,10 @@
   /pgf/decoration/.unknown/.code=%
     \pgfifdecoration{\pgfkeyscurrentname}{\edef\tikz@decoration@name{\pgfkeyscurrentname}}
     {\pgfifmetadecoration{\pgfkeyscurrentname}{\edef\tikz@decoration@name{\pgfkeyscurrentname}}
-    {\pgfkeys{/errors/unknown
-        key={/pgf/decoration/\pgfkeyscurrentname}{#1}}}},%
+    % fully expand \pgfkeyscurrentname before being passed to 
+    % `/errors/unknown key`
+    {\expandafter\pgfkeys\expandafter{\expanded{/errors/unknown
+        key={/pgf/decoration/\pgfkeyscurrentname}}{#1}}}},%
```

Comment:
Please introduce an additional macro à la
```
\long\def\pgfutil@expargs@Ne#1#2{%
  \expandafter#1\expandafter{\expanded{#2}}%
}
```
If we ever need some compatibility code for `\expanded` we only have to change it in one place.

Reply:
I think it would be good to start developing a pgfexpl module with drop-in replacements for expl3 macros. Once we have testsuite we can start refactoring stuff to use pgfexpl and some time in the far future switch to expl3 entirely.

Reply:
Just find expl3 already provides a general loader `exple3-generic.tex`, see lines starting from [`expl3.dtx`, line 958](https://github.com/latex3/latex3/blob/f8c2618e8fbbfa6aae84695093ba7ef042c35897/l3kernel/expl3.dtx#L958).

Reply:
Yes, indeed, that is what I was alluding to in my last comment but at this time I'm not prepared to switch.

Reply:
I didn't test it but wouldn't it be sufficient to use `.expand once` here to get the body of the message?

Reply:
@ilayn Not sufficient. `./expand once` will only expand the first `{` in `{/pgf/decoration/\pgfkeyscurrentname}{#1}`.

Reply:
`expand twice` then? or `expanded` if that's not enough? I mean we have a full fledged key system. it should be capable of such trivial tasks

Reply:
@ilayn A late thanks. Obviously I didn't get your point until Henri shown the code today.